public/app/plugins/datasource/tempo/datasource.test.ts

```diff
@@ -17,20 +23,50 @@ describe('Tempo data source', () => {
       })
     );
     const ds = new TempoDatasource(defaultSettings);
-    await expect(ds.query({ targets: [{ refId: 'refid1' }] } as any)).toEmitValuesWith((response) => {
-      const fields = (response[0].data[0] as DataFrame).fields;
-      expect(
-        fields.map((f) => ({
-          name: f.name,
-          values: f.values.toArray(),
-        }))
-      ).toMatchObject([
-        { name: 'serviceTags', values: [{ key: 'servicetag1', value: 'service' }] },
-        { name: 'logs', values: [{ timestamp: 12345, fields: [{ key: 'count', value: 1 }] }] },
-        { name: 'tags', values: [{ key: 'tag1', value: 'val1' }] },
-        { name: 'serviceName', values: ['service'] },
-      ]);
-    });
+    const response = await ds.query({ targets: [{ refId: 'refid1' }] } as any).toPromise();
```

Comment:
I changed this because `toEmitValuesWith` seems to mask actual stacktrace from inside the callback so if there is a bit more code or multiple matchers it does not help debugging. @hugohaggmark is it possible to propagate the error somehow? From the code it seems to me like the jest matchers aren't really meant to work that way.

Reply:
You're right, might need to look into that after v8. Thanks for mentioning it